alts/src/main/java/io/grpc/alts/FailingCallCredentials.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.alts;
+
+import com.google.common.base.Preconditions;
+import io.grpc.CallCredentials;
+import io.grpc.Status;
+import java.util.concurrent.Executor;
+
+/**
+ * {@code CallCredentials} that always fail the RPC.
+ */
+final class FailingCallCredentials extends CallCredentials {
```

Comment:
This class has no dependency on any alts classes and could be reusable by others. Just a suggestion to move it to `io.grpc` and mark it `@Internal` .

Reply:
I'd move to to `io.grpc.internal` to avoid polluting `io.grpc`'s namespace. But since there's no other user that just adds a dependency on an internal detail for no benefit. Every internal dependency contributes to version skew breakages, so I'd rather avoid it if there's no benefit. Once it is in io.grpc.internal it is also harder to notice when it is no longer used. It seems easy to move once there's a second user.

I think this is a hack as well, and so wouldn't want to encourage others to do the same. Using it seems strange for ComputeEngineChannelCredentials; why wouldn't the call credentials already be handling that themselves? For GoogleDefaultChannelCredentials it really seems like we should throw, as the call credential auto-detection can be incorrect and may need to be retried. But those sorts of changes seemed inappropriate in this PR; I tried to find the closest approximation of the previous behavior, now that we can't use interceptors.